src/views/datalist/data-list-directive.js

```diff
@@ -227,6 +279,80 @@ angular.module('patternfly.views').directive('pfDataList', function (pfUtils) {
             'Illegal use of pfDataList directive! ' +
             'Cannot allow both select box and click selection in the same data list.');
           }
+
+          if (!$scope.listId) {
+            $scope.listId = 'pf-data-list';
+          }
+
+          $scope.handleAction = function (actionName) {
+            var action = _.findWhere($scope.actions, {name: actionName});
+            if (action && action.actionFn && (action.isDisabled !== true)) {
+              action.actionFn(action, $scope.actionItem);
+            }
+          };
+
+          $scope.updateActions = function (item) {
+            $scope.actionItem = item;
+            if (typeof $scope.updateActionForItemFn === 'function') {
+              $scope.actions.forEach(function (action) {
+                $scope.updateActionForItemFn(action, item);
+              });
+            }
+          };
+
+          $scope.getActionHtml = function (action) {
+            var actionHtml = '';
+
+            if (action.isVisible !== false) {
+              actionHtml += '<li role=\'' + (action.isSeparator === true ? 'separator' : 'menuitem') + '\' ' +
+              'class="' + (action.isSeparator === true ? 'divider ' : '') + (action.isDisabled === true ? 'disabled' : '') + '"> ';
+              if (action.isSeparator !== true) {
+                actionHtml += '<a href="#" ' +
+                '   title="' + action.title + '" ' +
+                '   ng-click="handleAction(\'' + action.name + '\')"> ' +
+                ' ' + action.name +
+                '</a>';
```

Comment:
> The problem with that is that it would create the dropdown for every item in the list.

What about using `ng-if` and the aforementioned directive?